sdk/src/tx/reportTransactionError.ts

```diff
@@ -0,0 +1,96 @@
+import { Connection, SendTransactionError, VersionedTransactionResponse } from "@solana/web3.js";
+
+const getTransactionResult = async (txSig: string, connection: Connection): Promise<VersionedTransactionResponse> => {
+    return await connection.getTransaction(txSig, {
+        maxSupportedTransactionVersion: 0,
+    });
+};
+
+const getTransactionResultWithRetry = async (txSig: string, connection: Connection): Promise<VersionedTransactionResponse> => {
+    const start = Date.now();
+
+    const retryTimeout = 3_000; // Timeout after 3 seconds
+    const retryInterval = 800; // Retry with 800ms interval
+    const retryCount = 3; // Retry 3 times
+    
+    let currentCount = 0;
+    let transactionResult = await getTransactionResult(txSig, connection);
+
+    // Retry 3 times or until timeout as long as we don't have a result yet
+    while (!transactionResult && Date.now() - start < retryTimeout && currentCount < retryCount) {
+        transactionResult = await getTransactionResult(txSig, connection);
+        currentCount++;
+        // Sleep for 1 second
+        await new Promise(resolve => setTimeout(resolve, retryInterval));
+    }
```

Comment:
```suggestion
    // Retry 3 times or until timeout as long as we don't have a result yet
    while (!transactionResult && Date.now() - start < retryTimeout && currentCount < retryCount) {
        await new Promise(resolve => setTimeout(resolve, retryInterval));
        
        transactionResult = await getTransactionResult(txSig, connection);
        currentCount++;
    }
```

i'd sleep at the top so it doesnt do 2 consecutive calls